server/src/main/java/org/elasticsearch/search/aggregations/metrics/PercentilesConfig.java

```diff
@@ -165,7 +165,7 @@ public void parseExecutionHint(String executionHint) {
         }
 
         public TDigestExecutionHint getExecutionHint() {
-            return executionHint;
+            return executionHint == null ? TDigestExecutionHint.DEFAULT : executionHint;
```

Comment:
I think we should allow executionHint to be null here so that we can rely on the default defined in the new cluster setting.

Reply:
Right, this should only apply in cases where the executionHint hasn't been set through a call to a method that accesses the AggregationContext. In general, I think it's better to guarantee that this won't be used as null, otherwise we'd better use an Optional<TDigestExecutionHint> and modify all consumers accordingly. This does feel like overengineering, though - we have a default hint for this exact reason..